metaflow/plugins/logs_cli.py

```diff
@@ -0,0 +1,306 @@
+
+    def get_command(self, ctx, cmd_name):
+        if cmd_name not in self.list_commands(ctx):
+            # input from the CLI does not match a command, so we pass that
+            # as the args to the default command instead.
+            ctx.passed_cmd = cmd_name
+            cmd_name = self.default_cmd
+        return super(CustomGroup, self).get_command(ctx, cmd_name)
+
+    def parse_args(self, ctx, args):
+        # We first try to parse args as is, to determine whether we need to fall back to the default commmand
+        # if any options are supplied, the parse will fail, as the group does not support the options.
+        # In this case we fallback to the default command, inserting that as the first arg and parsing again.
+        # copy args as trying to parse will destroy them.
+        original_args = list(args)
+        try:
+            super().parse_args(ctx, args)
+            args_parseable = True
+        except Exception:
+            args_parseable = False
+        if not args or not args_parseable:
+            original_args.insert(0, self.default_cmd)
+        return super().parse_args(ctx, original_args)
+
+    def resolve_command(self, ctx, args):
+        cmd_name, cmd_obj, args = super(CustomGroup, self).resolve_command(ctx, args)
+        passed_cmd = getattr(ctx, "passed_cmd", None)
+        if passed_cmd is not None:
+            args.insert(0, passed_cmd)
+
+        return cmd_name, cmd_obj, args
+
+    def format_commands(self, ctx, formatter):
+        formatter = CustomFormatter(self.default_cmd, formatter)
+        return super(CustomGroup, self).format_commands(ctx, formatter)
+
+
+class CustomFormatter:
+    def __init__(self, default_cmd, original_formatter) -> None:
+        self.default_cmd = default_cmd
+        self.formatter = original_formatter
+
+    def __getattr__(self, name):
+        return getattr(self.formatter, name)
+
+    def write_dl(self, rows):
+        def _format(dup):
+            cmd, help = dup
+            if cmd == self.default_cmd:
+                cmd = cmd + " [Default]"
+            return (cmd, help)
+
+        rows = [_format(dup) for dup in rows]
+
+        return self.formatter.write_dl(rows)
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.group(cls=CustomGroup, help="Commands related to logs", default_cmd="show")
+@click.pass_context
+def logs(ctx):
+    # the logger is configured in cli.py
+    global echo
+    echo = ctx.obj.echo
+
+
+@logs.command(
+    help="Show stdout/stderr produced by a task or all tasks in a step. "
+    "The format for input-path is either <run_id>/<step_name> or "
+    "<run_id>/<step_name>/<task_id>."
+)
+@click.argument("input-path")
+@click.option(
+    "--stdout/--no-stdout",
+    default=False,
+    show_default=True,
+    help="Show stdout of the task.",
+)
+@click.option(
+    "--stderr/--no-stderr",
+    default=False,
+    show_default=True,
+    help="Show stderr of the task.",
+)
+@click.option(
+    "--both/--no-both",
```

Comment:
nit: I would add a comment (and in the next one) that this is overridden by specifying anything wrt stdout/stderr (I know it wasn't there before but unclear on the priority). Not suggesting changing the API but clarifying behavior.